Cabal/src/Distribution/PackageDescription/Check.hs

```diff
@@ -227,6 +227,7 @@ data CheckExplanation =
         | UnknownArch [String]
         | UnknownCompiler [String]
         | BaseNoUpperBounds
```

Comment:
Is there a need to leave `BaseNoUpperBounds` with your patch?

Reply:
It get's a different warning level, `PackageDistInexcusable` for base not having upper bounds, and `PackageDistSuspiciousWarn` for other dependencies.

I hoped that it'd just nag at the user for not having an upper bound on other dependencies then base.